src/psrsearch.jl

```diff
@@ -38,26 +36,25 @@ julia> z_n_binned([10., 0., 0., 0., 0.], 2)
 40.0
 ```
 """
-function z_n_binned(profile::Array{T,1}, n::Integer)::T where {T<:AbstractFloat}
+function z_n_binned(profile::AbstractVector{T}, n::Integer)::T where {T<:AbstractFloat}
     total = sum(profile)
     N = length(profile)
-    phase = Array(range(0, stop = N - 1) / N) * 2 * pi
+    phase = range(0, stop = N - 1) / N * 2 * pi
 
-    if total == 0
-        return 0
+    if iszero(total)
+        return zero(T)
     end
 
-    z = 0
+    z = zero(T)
     for k in range(1, stop = n)
-        s = 0
-        for i in range(1, stop = N)
-            s += profile[i] * sin(k * phase[i])
+        s = zero(T)
+        c = zero(T)
+        for i in eachindex(profile)
+            sk, ck = sincos(k * phase[i])
+            s += profile[i] * sk
+            c += profile[i] * ck
         end
         s = s^2
-        c = 0
-        for i in range(1, stop = N)
-            c += profile[i] * cos(k * phase[i])
-        end
         c = c^2
         z += c + s
```

Comment:
Sorry, I'm a bit confused here: isn't `z` simply the sum of `profile .^ 2`?  But perhaps I'm missing something :sweat_smile: